### PR TITLE
Add CIDR to URL Indicator

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [ ] In Progress
- [x] Ready

## Related Issues
Related: https://github.com/demisto/etc/issues/39650

## Description
Fixed an issue where some URLs (for example 255.255.255.255/25.com) are not recognized as an URL.
 
## Minimum version of Cortex XSOAR
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests

### CODE REVIEW

**Review by bakatzir — COMMENTED** [review]
please also add examples to the `Set invalid URLs` Playbook task in `Packs/Base/TestPlaybooks/playbook-reputation.json-Test.yml` to make sure we do not extract non URLs  with the editted regex.

**Review by bakatzir — APPROVED** [review]
I'm good with the changes.
let's wait for a 2nd reviewer. either @yuvalbenshalom or @yaakovi.

**Comment by ChanochShayner** [comment]
> I'm good with the changes.
> let's wait for a 2nd reviewer. either @yuvalbenshalom or @yaakovi.

@yuvalbenshalom @yaakovi Please, Thanks.

File: Packs/CommonTypes/ReleaseNotes/3_1_11.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+#### Indicator Types
+- **urlRep**: Fixed an issue where some URLs (for example 255.255.255.255/25.com) are not recognized as an URL.
```
Comment (yaakovi) [inline Packs/CommonTypes/ReleaseNotes/3_1_11.md:3]:
```suggestion
- **urlRep**: Fixed an issue where some URLs (for example 255.255.255.255/25.com) are not recognized as a URL.
```

File: Packs/CommonTypes/IndicatorTypes/reputation-url.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   "commitMessage": "",
   "shouldPublish": false,
   "shouldCommit": false,
-  "regex": "(?i)(?:(?:https?|ftp|hxxps?):(\\/?\\/?\\\\?\\\\?)|www\\[?\\.\\]?|ftp\\[?\\.\\]?)(?:[-_\\d\\p{L}]+\\[?\\.\\]?)+[-_\\d\\p{L}]+(?::\\d+)?(?:(?:((\\/)|(\\*)|(\\#)|(\\?)))+(?:[-\\d\\p{L}+&@'#\\/%=~_$!\\-:,.;\\\\*–—―|]*|\\([-\\d\\p{L}\\+&@#\\/%=~_\\$\\?!\\-:,\\.;\\\\\\*–—―\\|]*\\))*)*",
+  "regex": "(?i)((?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])(?:\\[\\.\\]|\\.)){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])(\\/)|(?:(?:https?|ftp|hxxps?):(\\/?\\/?\\\\?\\\\?)|www\\[?\\.\\]?|ftp\\[?\\.\\]?))(?:[-_\\d\\p{L}]+\\[?\\.\\]?)+[-_\\d\\p{L}]+(?::\\d+)?(?:(?:((\\/)|(\\*)|(\\#)|(\\?)))+(?:[-\\d\\p{L}+&@'#\\/%=~_$!\\-:,.;\\\\*–—―|]*|\\([-\\d\\p{L}\\+&@#\\/%=~_\\$\\?!\\-:,\\.;\\\\\\*–—―\\|]*\\))*)*",
```
Comment (yaakovi) [inline Packs/CommonTypes/IndicatorTypes/reputation-url.json:10]: I assume the slash should not be in a matched group.
are you intentionally supporting both `1.1.1.1` and `1[.]1[.]1[.]1`?
```suggestion
  "regex": "(?i)((?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])(?:\\[\\.\\]|\\.)){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])/|(?:(?:https?|ftp|hxxps?):(\\/?\\/?\\\\?\\\\?)|www\\[?\\.\\]?|ftp\\[?\\.\\]?))(?:[-_\\d\\p{L}]+\\[?\\.\\]?)+[-_\\d\\p{L}]+(?::\\d+)?(?:(?:((\\/)|(\\*)|(\\#)|(\\?)))+(?:[-\\d\\p{L}+&@'#\\/%=~_$!\\-:,.;\\\\*–—―|]*|\\([-\\d\\p{L}\\+&@#\\/%=~_\\$\\?!\\-:,\\.;\\\\\\*–—―\\|]*\\))*)*",
```

**Review by yaakovi — COMMENTED** [review]
what about the following (will they be a valid URL?):
* `https://1.1.1.1:5000/uri/suffix`
* `1.1.1.1/5/hello`
* `1.1.1.1/5`
* `http://1.1.1.1/5`

**Comment by ChanochShayner** [comment]
> what about the following (will they be a valid URL?):
> 
> * `https://1.1.1.1:5000/uri/suffix`
> * `1.1.1.1/5/hello`
> * `1.1.1.1/5`
> * `http://1.1.1.1/5`

image

**Comment by ChanochShayner** [comment]
@yaakovi 
Because there will be a standardization for URLs soon, and there will be changes, I just tried to fix the specific bug we got (and not add all the cases) and of course not to damage the existing Pattern.

About `1[.]1[.]1[.]1` - yes

https://1.1.1.1:5000/uri/suffix - yes
1.1.1.1/5/hello - no
1.1.1.1/5 - no
http://1.1.1.1/5 - yes
image

**Review by yaakovi — APPROVED** [review]

**Pull request merged**